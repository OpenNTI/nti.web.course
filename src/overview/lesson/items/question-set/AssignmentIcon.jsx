import './AssignmentIcon.scss';
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

LessonOverviewAssignmentIcon.propTypes = {
	assignment: PropTypes.object,
	assignmentHistory: PropTypes.object,
	large: PropTypes.bool
};
export default function LessonOverviewAssignmentIcon ({assignment, assignmentHistory, large}) {
	const now = new Date();
	const due = assignment.getDueDate();
	const noSubmit = assignment.isNonSubmit();
	const completed = assignmentHistory && assignmentHistory.Submission && assignmentHistory.Submission.getCreatedTime();
	const hasCompletedItem = assignment?.CompletedItem;
	const failed = assignment && assignment.CompletedItem && !assignment.CompletedItem.Success;

	let cls = '';

	if (noSubmit) { cls = 'no-submit'; }
	if (completed) { cls = completed > due ? 'late' : 'on-time'; }
	if (due && due < now) { cls = 'late'; }
	if (failed) { cls = 'failed'; }

	return (
		<div className={cx('lesson-overview-assignment-icon', cls, {large, ['no-completion']: completed && !hasCompletedItem})} />
	);
}
