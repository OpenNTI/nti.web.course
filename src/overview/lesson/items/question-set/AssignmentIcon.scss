@import '~nti-style-common';

.lesson-overview-assignment-icon {
	width: rem-calc(40px);
	height: rem-calc(40px);

	background-size: contain;
	background-image: url('../assets/assignment_due.png');

	&.ontime {
		background-image: url('../assets/assignment_ontime.png');
	}

	&.late {
		background-image: url('../assets/assignment_late.png');
	}
}
