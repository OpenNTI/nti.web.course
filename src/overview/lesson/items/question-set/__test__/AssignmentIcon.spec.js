import React from 'react';
import renderer from 'react-test-renderer';

import AssignmentIcon from '../AssignmentIcon';

/* eslint-env jest */
describe('Course overview assignment icon test', () => {
	test('Completed, late', async () => {
		// const due = assignment.getDueDate();
		// const noSubmit = assignment.isNonSubmit();
		// const completed = assignmentHistory && assignmentHistory.Submission && assignmentHistory.Submission.getCreatedTime();

		const assignment = {
			getDueDate: () => new Date('10/22/2017'),
			isNonSubmit: () => false
		};

		const assignmentHistory = {
			Submission: {
				getCreatedTime: () => new Date('10/24/2017')
			}
		};

		const cmp = renderer.create(
			<AssignmentIcon assignment={assignment} assignmentHistory={assignmentHistory}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Non submit', async () => {
		const assignment = {
			getDueDate: () => null,
			isNonSubmit: () => true
		};

		const cmp = renderer.create(
			<AssignmentIcon assignment={assignment}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Completed on time', async () => {
		const assignment = {
			getDueDate: () => Date.now(),
			isNonSubmit: () => true
		};

		const assignmentHistory = {
			Submission: {
				getCreatedTime: () => new Date('10/24/2017')
			}
		};

		const cmp = renderer.create(
			<AssignmentIcon assignment={assignment} assignmentHistory={assignmentHistory}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
