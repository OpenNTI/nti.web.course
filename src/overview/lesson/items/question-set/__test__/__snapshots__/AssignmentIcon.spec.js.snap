// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course overview assignment icon test Completed on time 1`] = `
<div
  className="lesson-overview-assignment-icon on-time"
/>
`;

exports[`Course overview assignment icon test Completed, late 1`] = `
<div
  className="lesson-overview-assignment-icon late"
/>
`;

exports[`Course overview assignment icon test Non submit 1`] = `
<div
  className="lesson-overview-assignment-icon no-submit"
/>
`;
